rotkehlchen/exchanges/kraken.py

```diff
@@ -782,3 +779,282 @@ def query_online_income_loss_expense(
             end_ts: Timestamp,  # pylint: disable=unused-argument
     ) -> List[LedgerAction]:
         return []  # noop for kraken
+
+    def query_pairs(self) -> None:
+        self.pairs = set(self.api_query('AssetPairs').keys())
+
+    def kraken_asset_to_pair(self, assets: List[Asset]) -> Optional[str]:
+        """Given a pair of assets find the kraken pair that has this two assets as
+        base and quote"""
+        if len(assets) != 2:
+            return None
+        assets_symbols = [asset.symbol for asset in assets]
+        pair = "".join(assets_symbols)
+        if pair not in self.pairs:
+            pair = "".join(assets_symbols[::-1])
+        if pair not in self.pairs:
+            # Try tansaliting one and not the other
+            assets_symbols = [
+                WORLD_TO_KRAKEN.get(assets[0].symbol, assets[0].symbol),
+                assets[1].symbol,
+            ]
+            pair = "".join(assets_symbols)
+            if pair not in self.pairs:
+                pair = "".join(assets_symbols[::-1])
+            if pair not in self.pairs:
+                # Try the other way around
+                assets_symbols = [
+                    WORLD_TO_KRAKEN.get(assets[1].symbol, assets[1].symbol),
+                    assets[0].symbol,
+                ]
+                pair = "".join(assets_symbols)
+                if pair not in self.pairs:
+                    pair = "".join(assets_symbols[::-1])
+                if pair not in self.pairs:
+                    # Try translation both
+                    assets_symbols = [
+                        WORLD_TO_KRAKEN.get(asset.symbol, asset.symbol)
+                        for asset in assets
+                    ]
+                    pair = "".join(assets_symbols)
+                    if pair not in self.pairs:
+                        pair = "".join(assets_symbols[::-1])
+                    if pair not in self.pairs:
+                        return None
+        return pair
+
+    def process_kraken_trades(
+        self,
+        raw_data: List[HistoryEvent],
+    ) -> List[Trade]:
+        try:
+            if len(self.pairs) == 0:
+                self.query_pairs()
+        except RemoteError as e:
+            self.msg_aggregator.add_error(f'Failed to query kraken pairs due to {str(e)}')
+            raise RemoteError from e
+        trades = []
+        get_attr = operator.attrgetter('event_identifier')
+        grouped_events = [list(g) for k, g in itertools.groupby(sorted(raw_data, key=get_attr), get_attr)]  # noqa: E501
+        for trade_parts in grouped_events:
+            is_spend_receive = any(
+                [trade.event_type == HistoryEventType.RECEIVE for trade in trade_parts],
+            )
+            if is_spend_receive and len(trade_parts) < 2:
```

Comment:
Will add also to the docs. This part of the logic only processes trades and what we have called `Instant trades` are a combination of spend/receive. If we are dealing with a spend/receive event the minimum required events are 2. Added this to be sure that the trades are processed correctly in case something is bad with the data in the database